rush-plugins/rush-amazon-s3-build-cache-plugin/src/test/AmazonS3Credentials.test.ts

```diff
@@ -0,0 +1,83 @@
+import {
+  AWS_ACCESS_KEY_ID,
+  AWS_SECRET_ACCESS_KEY,
+  AWS_SESSION_TOKEN,
+  fromAmazonEnv,
+  fromRushEnv
+} from '../AmazonS3Credentials';
+import { EnvironmentConfiguration } from '@rushstack/rush-sdk';
+
+describe('Amazon S3 Credentials', () => {
+  describe(fromAmazonEnv.name, () => {
+    afterEach(() => {
+      delete process.env[AWS_ACCESS_KEY_ID];
+      delete process.env[AWS_SECRET_ACCESS_KEY];
+      delete process.env[AWS_SESSION_TOKEN];
+    });
+
+    it('returns AWS vars when present in env', () => {
+      process.env[AWS_ACCESS_KEY_ID] = AWS_ACCESS_KEY_ID;
+      process.env[AWS_SECRET_ACCESS_KEY] = AWS_SECRET_ACCESS_KEY;
+      process.env[AWS_SESSION_TOKEN] = AWS_SESSION_TOKEN;
+
+      const credentials = fromAmazonEnv();
+
+      expect(credentials).toBeDefined();
+      expect(credentials!.accessKeyId).toBe(AWS_ACCESS_KEY_ID);
+      expect(credentials!.secretAccessKey).toBe(AWS_SECRET_ACCESS_KEY);
+      expect(credentials!.sessionToken).toBe(AWS_SESSION_TOKEN);
+    });
+
+    it('returns undefined sessionToken when not present', () => {
+      process.env[AWS_ACCESS_KEY_ID] = AWS_ACCESS_KEY_ID;
+      process.env[AWS_SECRET_ACCESS_KEY] = AWS_SECRET_ACCESS_KEY;
+
+      const credentials = fromAmazonEnv();
+
+      expect(credentials).toBeDefined();
+      expect(credentials!.accessKeyId).toBe(AWS_ACCESS_KEY_ID);
+      expect(credentials!.secretAccessKey).toBe(AWS_SECRET_ACCESS_KEY);
+      expect(credentials!.sessionToken).toBeUndefined();
+    });
+
+    it('returns undefined if access key and secret are not both present', () => {
```

Comment:
Would we want the plugin to throw instead if the env variables don't provide a complete credential?